File: python/leetcode/two_sum.py
# https://leetcode.com/problems/two-sum/
from typing import List

# Solution 1


def twoSum_1(nums: list[int], target: int) -> list[int]:
    for i in range(0, len(nums)):
        for j in range(i+1, len(nums)):
            if target == nums[i]+nums[j]:
                return [i, j]
    return null

# Solution 2


def twoSum_2(nums: list[int], target: int):
    d = {}
    for i, num in enumerate(nums):
        if target-num in d:
            return [d[target-num], i]
        d[num] = i
    return [0, 0]


nums = [2, 7, 11, 15]
target = 9
result1 = twoSum_1(nums, target)
result2 = twoSum_2(nums, target)

print(result1, result2)

File: csharp/leetcode/two_sum.cs
// https://leetcode.com/problems/two-sum/
public class Two_Sum
{
    //Solution 1
    public int[]? TwoSum1(int[] nums, int target)
    {
        for (int i = 0; i < nums.Length; i++)
        {
            for (int j = i + 1; j < nums.Length; j++)
            {
                if (i != j)
                {
                    int value = nums[i] + nums[j];
                    if (value == target)
                    {
                        return new int[] { i, j };
                    }
                }
            }
        }
        return null;
    }

    //Solution 2
    public int[]? TwoSum2(int[] nums, int target)
    {
        Dictionary<int, int> result = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            if (result.ContainsKey(target - nums[i]))
            {
                return new int[] { i, result[target - nums[i]] };
            }
            result.Add(nums[i], i);
        }
        return new int[] { 0, 0 };
    }
}